roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -2252,31 +2252,21 @@ RooDataSet *RooAbsPdf::generate(const RooArgSet &whatVars, const RooDataSet& pro
 
 Int_t* RooAbsPdf::randomizeProtoOrder(Int_t nProto, Int_t, Bool_t resampleProto) const
 {
-  // Make unsorted linked list of indeces
-  RooLinkedList l ;
-  Int_t i ;
-  for (i=0 ; i<nProto ; i++) {
-    l.Add(new RooInt(i)) ;
-  }
-
   // Make output list
   Int_t* lut = new Int_t[nProto] ;
 
   // Randomly samply input list into output list
   if (!resampleProto) {
     // In this mode, randomization is a strict reshuffle of the order
-    for (i=0 ; i<nProto ; i++) {
-      Int_t iran = RooRandom::integer(nProto-i) ;
-      RooInt* sample = (RooInt*) l.At(iran) ;
-      lut[i] = *sample ;
-      l.Remove(sample) ;
-      delete sample ;
+    std::iota(lut, lut + nProto, 0); // fill the vector with 0 to nProto - 1
+    // shuffle code taken from https://en.cppreference.com/w/cpp/algorithm/random_shuffle
+    // (std::random_shuffle was deprecated in C++17)
+    for (int i = nProto-1; i > 0; --i) {
+        std::swap(lut[i], lut[RooRandom::integer(i+1)]);
     }
   } else {
     // In this mode, we resample, i.e. events can be used more than once
-    for (i=0 ; i<nProto ; i++) {
-      lut[i] = RooRandom::integer(nProto);
-    }
+    std::generate(lut, lut + nProto, [&](){ return RooRandom::integer(nProto); });
```

Comment:
the `()` in the lambda expression is not needed (depending on your coding style you might want it anyway)

Reply:
Wow I didn't know! Thanks!